go/client/install_osx.go

```diff
@@ -0,0 +1,568 @@
+		installAction = keybase1.InstallAction_REINSTALL
+		installStatus = keybase1.InstallStatus_INSTALLED
+	}
+
+	return installStatus, installAction, keybase1.Status{Name: "OK"}
+}
+
+type ServicesStatus struct {
+	Service []keybase1.ServiceStatus `json:"service"`
+	Kbfs    []keybase1.ServiceStatus `json:"kbfs"`
+}
+
+func serviceStatusFromLaunchd(ls launchd.Service, infoPath string) (keybase1.ServiceStatus, bool) {
+	status := keybase1.ServiceStatus{
+		Label: ls.Label(),
+	}
+
+	st, err := ls.LoadStatus()
+	if err != nil {
+		status.Status = errorStatus("LAUNCHD_ERROR", err.Error())
+		return status, true
+	}
+
+	if st == nil {
+		status.InstallStatus = keybase1.InstallStatus_NOT_INSTALLED
+		status.InstallAction = keybase1.InstallAction_INSTALL
+		status.Status = keybase1.Status{Name: "OK"}
+		return status, true
+	}
+
+	status.Label = st.Label()
+	status.Pid = st.Pid()
+	status.LastExitStatus = st.LastExitStatus()
+
+	// Check service info file (if present) and if the service is running (has a PID)
+	var serviceInfo *libkb.ServiceInfo
+	if infoPath != "" {
+		if status.Pid != "" {
+			serviceInfo, err = libkb.WaitForServiceInfoFile(infoPath, status.Pid, 20, 200*time.Millisecond, "service status")
+			if err != nil {
+				status.InstallStatus = keybase1.InstallStatus_ERROR
+				status.InstallAction = keybase1.InstallAction_REINSTALL
+				status.Status = errorStatus("LAUNCHD_ERROR", err.Error())
+				return status, true
+			}
+		}
+		if serviceInfo != nil {
+			status.Version = serviceInfo.Version
+		}
+	}
+
+	if status.Pid == "" {
+		status.InstallStatus = keybase1.InstallStatus_ERROR
+		status.InstallAction = keybase1.InstallAction_REINSTALL
+		status.Status = errorStatus("LAUNCHD_ERROR", fmt.Sprintf("%s is not running", st.Label()))
+		return status, true
+	}
+
+	return status, false
+}
+
+func serviceStatusesFromLaunchd(ls []launchd.Service) []keybase1.ServiceStatus {
+	c := []keybase1.ServiceStatus{}
+	for _, l := range ls {
+		s, done := serviceStatusFromLaunchd(l, "")
+		if !done {
+			s.Status = keybase1.Status{Name: "OK"}
+		}
+		c = append(c, s)
+	}
+	return c
+}
+
+func listServices() (*ServicesStatus, error) {
+	services, err := launchd.ListServices([]string{"keybase.service", "homebrew.mxcl.keybase"})
+	if err != nil {
+		return nil, err
+	}
+	kbfs, err := launchd.ListServices([]string{"keybase.kbfs.", "homebrew.mxcl.kbfs"})
+	if err != nil {
+		return nil, err
+	}
+
+	return &ServicesStatus{
+		Service: serviceStatusesFromLaunchd(services),
+		Kbfs:    serviceStatusesFromLaunchd(kbfs)}, nil
+}
+
+func showServices(out io.Writer) error {
+	err := launchd.ShowServices([]string{"keybase.service", "homebrew.mxcl.keybase"}, "Keybase", out)
+	if err != nil {
+		return err
+	}
+	err = launchd.ShowServices([]string{"keybase.kbfs.", "homebrew.mxcl.kbfs"}, "KBFS", out)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func DiagnoseSocketError(ui libkb.UI, err error) {
+	t := ui.GetTerminalUI()
```

Comment:
This is only meant for people to diagnose from command line, so I think TerminalUI is good here?